src/cmd.rs

```diff
@@ -0,0 +1,367 @@
+use std::ffi;
+use std::fmt;
+use std::io::Write;
+use std::io;
+use std::process;
+use std::str;
+
+use failure;
+
+/// Extend `Command` with helpers for running the current crate's binaries.
+pub trait CommandCargoExt {
+    /// Create a `Command` to run the crate's main binary.
+    ///
+    /// # Examples
+    ///
+    /// ```rust,ignore
+    /// extern crate assert_cli;
+    /// use std::process::Command;
+    /// use assert_cli::cmd::*;
+    ///
+    /// Command::main_binary()
+    ///     .output()
+    ///     .unwrap();
+    /// ```
+    fn main_binary() -> Self;
+
+    /// Create a `Command` Run a specific binary of the current crate.
+    ///
+    /// # Examples
+    ///
+    /// ```rust,ignore
+    /// extern crate assert_cli;
+    /// use std::process::Command;
+    /// use assert_cli::cmd::*;
+    ///
+    /// Command::cargo_binary("assert_fixture")
+    ///     .output()
+    ///     .unwrap();
+    /// ```
+    fn cargo_binary<S: AsRef<ffi::OsStr>>(name: S) -> Self;
+}
+
+impl CommandCargoExt for process::Command {
+    fn main_binary() -> Self {
+        let mut cmd = process::Command::new("carg");
```

Comment:
`carg` -> `cargo`?
Same in [line 51](https://github.com/assert-rs/assert_cli/pull/98/files#diff-518bd9db9fa60ba5991cb1df6489cb2fR51)

Reply:
Thanks!

Obviously very little testing has been done which might be clear considering this is here :).  This is a rough sketch to get high level feedback before actually getting the details right.